### PR TITLE
MergeObjects Plugin

### PR DESCRIPTION
**Pull request by AetherUnbound**
This module fills holes on a per-object basis, essentially preserving object labels in hole filling. This is one of two PRs to address Issue #3369. This merges smaller labeled objects into a neighboring object based on the object that contributes the most pixels to said object's border. 

Before merge: 
image

After merge:
image

### CODE REVIEW

**Comment by AetherUnbound** [comment]
Since it seems there already exists a `SplitOrMergeObjects` module, does this module duplicate functionality? If it employs a different algorithm, is it useful to add as a different approach? I can also convert this module to operate on `Objects` @mcquin.

**Comment by AnneCarpenter** [comment]
I can't answer your question but just wanted to note that IIUC, I *think* the functionality you are describing is available in IdentifyPrimaryObjects... but only for 2D. So it would not be shocking if you needed to add this functionality in the absence of IdentifyPrimaryObjects being upgraded to 3D (which is beastly).

**Comment by mcquin** [comment]
> Since it seems there already exists a `SplitOrMergeObjects` module, does this module duplicate functionality? If it employs a different algorithm, is it useful to add as a different approach?

The [merge options](https://github.com/CellProfiler/CellProfiler/blob/master/cellprofiler/modules/splitormergeobjects.py#L137-L150) for `SplitOrMergeObjects` indicate different approaches for merging objects than this implementation. Until that module supports 3D, it's sensible to leave this as a separate module.

> I can also convert this module to operate on `Objects`.

Sounds great. :+1:

File: cellprofiler/modules/mergelabels.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# coding=utf-8
+
+"""
+MergeLabels
+===========
+
+**MergeLabels** merges labels below a certain threshold into its most prevalent, adjacent neighbor.
```
Comment (mcquin) [inline cellprofiler/modules/mergelabels.py]: I think you should add a note here stating that objects below the threshold with no adjacent neighbors will be removed.

**Comment by AetherUnbound** [comment]
Per @AnneCarpenter's request in #3381 I've moved this into the plugins folder. This is probably ready to be merged.

**Comment by AetherUnbound** [comment]
I'm going to close this PR and submit a PR to CellProfiler-Plugins instead (didn't realize that was a repo!).

**Pull request closed without merging**